src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -317,6 +317,19 @@ private function hasUserChanged(UserInterface $user)
             return true;
         }
 
+        $userRoles = array_map('strval', (array) $user->getRoles());
+        $rolesChanged =
+            (bool) \count(
+                array_merge(
+                    array_diff($this->getRoleNames(), $userRoles),
+                    array_diff($userRoles, $this->getRoleNames())
+                )
+            );
+
+        if ($rolesChanged) {
```

Comment:
Could be shortened to something like `count($userRoles) !== count($this->getRoleNames()) || count($userRoles) !== count(array_intersect($userRoles, $this->getRoleNames())`

Reply:
thanks!